Sources/AblyChat/Message.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+public typealias MessageHeaders = Headers
+public typealias MessageMetadata = Metadata
+
+public struct Message: Sendable {
+    public var timeserial: String
+    public var clientID: String
+    public var roomID: String
+    public var text: String
+    public var createdAt: Date
+    public var metadata: MessageMetadata
+    public var headers: MessageHeaders
+
+    public init(timeserial: String, clientID: String, roomID: String, text: String, createdAt: Date, metadata: MessageMetadata, headers: MessageHeaders) {
+        self.timeserial = timeserial
+        self.clientID = clientID
+        self.roomID = roomID
+        self.text = text
+        self.createdAt = createdAt
+        self.metadata = metadata
+        self.headers = headers
+    }
+
+    public func isBefore(_: Message) -> Bool {
```

Comment:
Maybe it will be more idiomatic to make `Message` implement `Comparable` protocol?

Reply:
I’m not sure. The thing is that if we were to implement `Comparable` as suggested — that is, using only the message’s `timeserial`, then that would mean that we’d also be implementing `Equatable`, and that we’d consider two messages to be equal if their `timeserial`s are equal. I’m not sure that’s the `Equatable` behaviour we’d wish to have — I think you’d either want to compare the messages member-by-member.

What we maybe _could_ do is change `Message.timeserial` into a type `Timeserial`. The maybe we could implement `Comparable` on `Timeserial`, but I don’t know anything about timeserials and whether they implement a strict total order, which is a requirement for `Comparable`.

Reply:
I totally agree with your points, but the methods `isBefore`, `isAfter`, and `isEqual`, which are not the same as `<`, `>`, and `==`, can be confusing. Maybe it’s better to remove them from the struct and provide them as extensions when needed?

Reply:
What do you mean by "provide them as extensions when needed"?

Reply:
I would argue that `message1 > message2` is the same as `message2.isBefore(message1)`. This comparison with operators looks confusing.

Reply:
I'm leaning towards @ttypic's suggestion of removing this until we know we need this. However, I also think using `>` and `<` in the this context is confusing... 

We can add an extension to provide conformance to `Equatable` later e.g. taken from [Apple docs](https://developer.apple.com/documentation/swift/equatable#Conforming-to-the-Equatable-Protocol)
```

extension StreetAddress: Equatable {
    static func == (lhs: StreetAddress, rhs: StreetAddress) -> Bool {
        return
            lhs.number == rhs.number &&
            lhs.street == rhs.street &&
            lhs.unit == rhs.unit
    }
} 
```

We can also define whatever logic we want to deem two messages equal (whether that's member my member comparison or something involving timestamps)

Reply:
I’d prefer, within this task, not to pick and choose which functionality offered by the JS public API we’re going to add to our public API. If the functionality itself doesn't make sense, then let's sort that out in JS first. If the functionality makes sense but doesn't seem right for Swift, let's either figure out the right Swift API here, or keep it as it is and make an issue to come back to it later.

Reply:
> by "first" I mean "let's stick with this and revisit once it's sorted in JS", not "let's block the current PR until it's sorted in JS"

I'm fine with this given this is just the API task and we shouldn't already be held up... we're also going to have a lot more PRs coming in so it's an easy change to make after. We should however, keep a decision log/table of sorts to know what we need to come back to... I'll action this later today

Reply:
I’d like to get this conversation resolved so we can merge — what needs doing here @umair-ably?

Reply:
for this in particular, nothing. they're already throwing fatalErrors/not implemented yet so we can discuss if needed when we revisit the implementation - happy to resolve for now